### PR TITLE
Writing server tests

### PR DESCRIPTION
**Pull request by PaulKovalov**
Here I add a few tests, I will add more later to test more specific edge cases

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BadRequestException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.google.sps.servlets;
```
Comment (PaulKovalov) [inline portfolio/src/main/java/com/google/sps/servlets/BadRequestException.java:1]: The diff seems to be broken here - I don't know why this file appeared in the diff, I have it in base branch as well. `git diff server-testing writing-server-tests` doesn't show this file.

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -37,6 +38,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String username = request.getParameter("username");
     String text = request.getParameter("text");
+    // if any of the necessary fields are absent, raise an error
+    if (username == null || text == null || text.trim().length() == 0) {
```
Comment (harry-7) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:42]: I suggest making this into different Ifs and give corresponding error message, (invalid user name, no text etc). This makes it easier to test and we can test each condition.

File: portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java [hunk 3]
```diff
@@ -26,16 +36,84 @@ public void setUp() {
...
+      JsonObject jsonObject = (new JsonParser()).parse(responseString).getAsJsonObject();
+      assertEquals(jsonObject.get("username").toString(), "\"Paul\"");
+      assertEquals(jsonObject.get("text").toString(), "\"A nice comment\"");
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithInvalidPayload_v1() {
```
Comment (harry-7) [inline portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:57]: Same comment as the other PR. Have elaborative test names.

File: portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java [hunk 4]
```diff
@@ -26,16 +43,127 @@ public void setUp() {
...
+    try {
+      servlet.doPost(request, response);
+      assertEquals(response.getStatus(), BAD_REQUEST);
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithInvalidCommentTextValue() {
```
Comment (harry-7) [inline portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:118]: suggest making these tests named testCreateCommentFailInvalidCommentTextValue() and so on

File: portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java [hunk 5]
```diff
@@ -26,16 +43,127 @@ public void setUp() {
...
     request.setMethod("POST");
     request.setContentType("text/html");
-    request.addParameter("username", "paul");
+    request.addParameter("username", "Paul");
+    // set the username and text full of spaces
+    request.addParameter("text", "        ");
```
Comment (harry-7) [inline portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:123]: I am not aware of the other PRs so I don't know how you implemented this. But checking for white space on backend seems like unnecessary overhead. You can very easily check this on frontend itself and not let the POST happen in the first place.

File: portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java (hunk 4 above)
Comment (PaulKovalov) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:118]: You mean I have to specify the expected condition like fail/succes?

Comment (harry-7) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:118]: Yes. the current name seems to mean like you can create with invalid comment text value which is not the case.

File: portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java (hunk 5 above)
Comment (PaulKovalov) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:123]: My frontend does not check for empty input because it uses plain HTML form

Comment (harry-7) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:123]: You can definitely do it even with HTML5 and java script together. Check out https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#Validating_forms_using_JavaScript for example. Infact, you can even write your own custom validator in js.

Comment (PaulKovalov) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:123]: Thank you, didnt know about this

Comment (PaulKovalov) [reply portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java:123]: So do you suggest to remove this test as well as check for empty input from servlet?
